src/Queue/QueueBase.php

```diff
@@ -71,34 +71,32 @@
    */
   public function __construct($name, Connection $connection,
     ModuleHandlerInterface $modules, LoggerInterface $logger) {
-    // Check our active storage to find the full config
+    $this->options = ['name' => $name];
+
+    // Check our active storage to find the the queue config
     $config = \Drupal::config('rabbitmq.config');
     $queues = $config->get('queues');
-
-    if (!$queues && !isset($queues[$name])) {
-      $logger->error('Cannot find queue information in active storage %yml for queue %name', array('%yml' => 'rabbitmq.queues', '%name' => $name));
-      // We should probably throw an error instead?
-      return;
+    if ($queues && isset($queues[$name])) {
+      $this->options += $queues[$name];
     }
 
-    $this->options = $queues[$name];
     $this->name = $name;
     $this->connection = $connection;
     $this->logger = $logger;
     $this->modules = $modules;
 
-    // Declare any exchanges required
+    // Declare any exchanges required if configured
     $exchanges = $config->get('exchanges');
     if ($exchanges) {
       foreach ($exchanges as $name => $exchange) {
         $this->getChannel()->exchange_declare(
           $name, 
-          $exchange['type'],
-          $exchange['passive'],
-          $exchange['durable'],
-          $exchange['auto_delete'],
-          $exchange['internal'],
-          $exchange['nowait']
+          isset($exchange['type']) ? $exchange['type'] : 'direct',
+          isset($exchange['passive']) ? $exchange['passive'] : FALSE,
+          isset($exchange['durable']) ? $exchange['durable'] : TRUE,
+          isset($exchange['auto_delete']) ? $exchange['auto_delete'] : FALSE,
+          isset($exchange['internal']) ? $exchange['internal'] : FALSE,
+          isset($exchange['nowait']) ? $exchange['nowait'] : FALSE
```

Comment:
I think this is config rather than code so should be **only** in the yml file ... and we already provide a yml in the contrib. I think is more clean in this way...